error-handling.js

```diff
@@ -3,17 +3,35 @@ exports.handleInvalidPaths = (req, res) => {
 };
 
 exports.handleCustomErrors = (err, req, res, next) => {
-  //console.log(err + "<<<<<<<<<");
+  console.log(err);
+  // if (err.msg === `article_id does not exist` && err.code === "22P02") {
+  //   res.status(400).send({ msg: "Invalid article ID" });
+  // }
   if (err.status && err.msg) {
     res.status(err.status).send({ msg: err.msg });
   } else {
     next(err);
   }
 };
 
+exports.handleArticle404 = (err, req, res, next) => {
+  console.log(err);
+  if (err.code === "23503") {
+    //foreign key restraint
+    console.log("23503");
+    res.status(404).send({ msg: `article ID does not exist` }); // need to also handle situation where username does not exist
+  } else {
+    next(err);
+  }
+};
+
```

Comment:
Not sure you need this error handler as your error handler on line 10 should do this for you. But happy to approve anyway :)